gio/src/task.rs

```diff
@@ -59,11 +60,58 @@ impl Task {
             ffi::g_task_set_priority(self.to_glib_none().0, priority.into_glib());
         }
     }
+}
+
+pub trait TaskExtManual<T: ValueType> {
+    #[doc(alias = "g_task_return_value")]
+    fn return_value(&self, result: &T);
+
+    #[doc(alias = "g_task_propagate_value")]
+    fn propagate_value(&self) -> Result<T, glib::Error>;
+}
+
+impl<T: ValueType> TaskExtManual<T> for Task {
+    fn return_value(&self, result: &T) {
+        unsafe extern "C" fn value_free(value: *mut c_void) {
+            let _: glib::Value = from_glib_full(value as *mut glib::gobject_ffi::GValue);
+        }
 
+        unsafe {
+            ffi::g_task_return_pointer(
+                self.to_glib_none().0,
+                result.to_value().to_glib_full() as *mut _,
+                Some(value_free),
+            )
+        }
+    }
+
+    fn propagate_value(&self) -> Result<T, glib::Error> {
+        unsafe {
+            let mut error = ptr::null_mut();
+            let value = ffi::g_task_propagate_pointer(self.to_glib_none().0, &mut error);
+            if !error.is_null() {
+                return Err(from_glib_full(error));
+            }
+            let value =
+                Option::<glib::Value>::from_glib_full(value as *mut glib::gobject_ffi::GValue)
+                    .unwrap_or_else(|| glib::Value::from_type(glib::types::Type::UNIT));
+            Ok(T::from_value(&value))
+        }
+    }
+}
+
+pub trait TaskExtManualSend<T: ValueType + Send>: TaskExtManual<T> {
```

Comment:
These two traits can be in a single one btw, you can make the functions generic over `T: ValueType` vs `T: ValueType + Send`